src/OpenToolkit.Mathematics/Color/Color4.cs

```diff
@@ -0,0 +1,1095 @@
+        [Pure]
+        public override string ToString()
+        {
+            return $"({X}, {Y}, {Z}, {W})";
+        }
+
+        /// <summary>
+        /// Compares whether this Color4 structure is equal to the specified object.
+        /// </summary>
+        /// <param name="obj">An object to compare to.</param>
+        /// <returns>True obj is a Color4 structure with the same components as this Color4; false otherwise.</returns>
+        [Pure]
+        public override bool Equals(object obj)
+        {
+            return obj is Color4<T> other && Equals(other);
+        }
+
+        /// <summary>
+        /// Calculates the hash code for this Color4 structure.
+        /// </summary>
+        /// <returns>A System.Int32 containing the hashcode of this Color4 structure.</returns>
+        [Pure]
+        public override int GetHashCode()
+        {
+            unchecked
+            {
+                var hashCode = X.GetHashCode();
+                hashCode = (hashCode * 397) ^ Y.GetHashCode();
+                hashCode = (hashCode * 397) ^ Z.GetHashCode();
+                hashCode = (hashCode * 397) ^ W.GetHashCode();
+                return hashCode;
+            }
+        }
+    }
+
+    /// <summary>
+    /// Factory for conversion between color types.
+    /// </summary>
+    public static class Color4
+    {
+        /// <summary>
+        /// Adds all the elements of left to all the elements of right.
+        /// </summary>
+        /// <param name="left">The left hand operand.</param>
+        /// <param name="right">The right hand operand.</param>
+        /// <returns>The element generated by adding the two colors together.</returns>
+        public static Color4<Rgba> Add(this in Color4<Rgba> left, in Color4<Rgba> right) =>
+            new Color4<Rgba>(left.X + right.X, left.Y + right.Y, left.Z + right.Z, left.W + right.W);
+
+        /// <summary>
+        /// Subtracts all the elements of right from all the elements of left.
+        /// </summary>
+        /// <param name="left">The left hand operand.</param>
+        /// <param name="right">The right hand operand.</param>
+        /// <returns>The element generated by subtracting all elements of right with all elements from left.</returns>
+        public static Color4<Rgba> Subtract(this in Color4<Rgba> left, in Color4<Rgba> right) =>
+            new Color4<Rgba>(left.X - right.X, left.Y - right.Y, left.Z - right.Z, left.W - right.W);
+
+        /// <summary>
+        /// Multiplies all the elements of left with all the elements of right.
+        /// </summary>
+        /// <param name="left">The left hand operand.</param>
+        /// <param name="right">The right hand operand.</param>
+        /// <returns>The element generated by multiplying the two colors together.</returns>
+        public static Color4<Rgba> Multiply(this in Color4<Rgba> left, in Color4<Rgba> right) =>
```

Comment:
Multiplying colors that way seems odd, since various color space-theories can be applied here..... and its mostly a matter of blending, neh? additive/subtractive/godknowswhat. 

Id leave `Multiply` out here, same for Add/Sub/Div, but I am also no expert of colortheory (Ideally some other class could do the color mixing stuff depending on whateverisnecessaryorneededtodoso.

I also didnt pay attention if its done in Color3 as well, if so, please remove there as well.

Reply:
I dont really get what the problem here is? this is available on almost any color library out there, you need ways of mixing colors. And these operators are the most standard, we should even come back to add more later on.

Reply:
But isnt mixing colors done with functions like `Lerp` then? Can you name another library which does Add/Div/Mul/Sub on colors? I really am just curious, this is not an attack :)

Reply:
Lerp is not mixing colors, lerp is... lerping between them
![image](https://user-images.githubusercontent.com/20968096/75193844-58b13080-5757-11ea-8334-8844e6b3a8e7.png)
This is my go-to paint program, and these are all the functions for mixing colors you can use.
also the reason we are doing this as methods and not as operators is because all of these make sense in the RGBA color space only.
https://documentation.help/NI-Vision-Builder-AI-Vision-Assistant/ColorOp_Controls.html
https://docs.microsoft.com/en-us/dotnet/api/system.windows.media.color.op_addition?view=netframework-4.8

Reply:
Albeit those are just labels on a control, i was curious as to how its implemented underneath and it seems its just that, divide, divides, add adds ... I honestly never bothered to look that up before <https://en.wikipedia.org/wiki/Blend_modes> TIL.

Reply:
Ah I think i also remember why I was thinking about commenting on those methods again... as soon as alpha is involved its some sort of Blending, and blending is a different story, isnt it? <https://en.wikipedia.org/wiki/Alpha_compositing>